app/scripts/components/datasets/s-explore/timeseries/trigger-rect.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { useEffect, useMemo, useRef } from 'react';
+import { select } from 'd3';
+
+import { useTimeseriesContext } from './context';
+
+type TriggerRectProps = {
+  onDataOverOut: (result: { hover: boolean; date: Date | null }) => void;
+  onDataClick: (result: { date: Date }) => void;
+};
+
+type Point = {
+  x: number;
+  y: number;
+};
+
+type HotZone = Point & {
+  date: Date;
+  radius: number;
+};
+
+function dist(p1: Point, p2: Point) {
+  return Math.sqrt(Math.pow(p2.x - p1.x, 2) + Math.pow(p2.y - p1.y, 2));
+}
+
+function getHotZone(zones: HotZone[], mouse) {
+  return zones.find(
+    (z) => dist(z, { x: mouse.layerX, y: mouse.layerY }) <= z.radius
+  );
+}
+
+export default function TriggerRect(props: TriggerRectProps) {
+  const { onDataOverOut, onDataClick } = props;
+  const { width, height, data, x, zoomXTranslation, zoomBehavior } =
+    useTimeseriesContext();
+  const elRef = useRef<SVGRectElement>(null);
+  const hoverDataRef = useRef(false);
+
+  // Since we're applying the zoom behavior on an invisible trigger rect, it
+  // swallows all mouse events. To be able to handle a hover state or even a
+  // click on a data point we create a list of coordinates for where the data
+  // is, and then check if a mouse event is within a given radius of them.
+  const dataHotZones = useMemo(() => {
```

Comment:
I thought there must be some easy css tricks using `pointer-events` but wow it seems surprisingly tricky to propagate events from the element with higher z-index to lower one!